spec/indexSpec.js

```diff
@@ -48,15 +64,68 @@ describe("AssetFingerprint", function() {
 
       it("should use value when provided", function () {
         var dir = 'config/initializers';
-        fingerprint = new AssetFingerprint(dir, true, "FINGERPRINT_VALUE");
-        expect(fingerprint.fingerprintName).toEqual("FINGERPRINT_VALUE");
+        fingerprint = new AssetFingerprint(dir, true, "TEST_FINGERPRINT");
+        expect(fingerprint.fingerprintName).toEqual("TEST_FINGERPRINT");
+      });
+    });
+
+    describe("fingerprint name", function() {
+      beforeEach(function() {
+        spyOn(fs, "existsSync").and.returnValue(true);
+        spyOn(fs, "readFileSync").and.returnValue("TEST_FINGERPRINT");
+      });
+
+      it("should validate for duplicate name.", function () {
+        var dir = 'config/initializers';
+        fingerprint = function() { new AssetFingerprint(dir, true, "TEST_FINGERPRINT"); }
+        expect(fingerprint).toThrowError('The provided fingerprint name has already been used.');
+      });
+
+      it("should validate the file name characters.", function() {
+        var dir = 'config/initializers';
+        fingerprint = function() { new AssetFingerprint(dir, true, "asset_FINGERPRINT"); }
+        expect(fingerprint).toThrowError('Please supply a fingerprint name that is all caps separating words by underscores (i.e. CUSTOM_ASSET_FINGERPRINT).');
+      });
+
+      it("should validate the file name format.", function() {
+        var dir = 'config/initializers';
+        fingerprint = function() { new AssetFingerprint(dir, true, "BAD_VALUE"); }
+        expect(fingerprint).toThrowError('Please supply a fingerprint name that is all caps separating words by underscores (i.e. CUSTOM_ASSET_FINGERPRINT).');
+      });
+    });
+
+    describe("fingerprint name", function() {
+      beforeEach(function() {
+        spyOn(fs, "existsSync").and.returnValue(true);
+        spyOn(fs, "readFileSync").and.returnValue("ASSET_FINGERPRINT");
+      });
+
+      it("should not validate for duplicate name with default arguments.", function () {
+        var dir = 'config/initializers';
+        fingerprint = new AssetFingerprint(dir);
+        expect(fingerprint.fingerprintName).toEqual("ASSET_FINGERPRINT");
+      });
+    });
+
+    describe("fingerprint name", function() {
```

Comment:
Honestly just curious, but in the case where there are different setups for the same describe (like in this case where we have 3 "fingerprint name" blocks) is this an idiomatic pattern, vs having them nested under one block like below?

```javascript
describe("fingerprint name", function(){
  describe("valid", function(){
    beforeEach(function() {
      ...
    });

    it("should use value when provided", function() {
      ...
    });
  });


  describe("invalid", function(){
    beforeEach(function() {
      ...
    });

    it("should validate for duplicate name", function() {
      ...
    });
  });
});
```

Reply:
The return values setup for the spyOn calls for fs are different and must be setup in a new describe section.  There is one test I can move up with the others that has the same criteria.